packages/application-shell/src/components/route-catch-all/route-catch-all.js

```diff
@@ -1,51 +1,38 @@
 import { Route } from 'react-router-dom';
-import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
 import { PageNotFound } from '@commercetools-frontend/application-components';
 
-export class ForcePageReload extends React.PureComponent {
-  static displayName = 'ForcePageReload';
-  // The `forcedReload` ensures that the page is always reloaded
-  // from the server, instead of from the browser cache.
-  // https://developer.mozilla.org/en-US/docs/Web/API/Location/reload
-  reloadPage = () => window.location.reload(/* forcedReload */ true);
-  componentDidMount() {
-    this.reloadPage();
-  }
-  render() {
-    return null;
-  }
-}
+const ForcePageReload = () => {
+  React.useEffect(() => {
+    window.location.reload(/* forcedReload */ true);
+  }, []);
+  return null;
+};
 
-export class RouteCatchAll extends React.PureComponent {
-  static displayName = 'RouteCatchAll';
-  static propTypes = {
-    servedByProxy: PropTypes.bool,
-  };
+const RouteCatchAll = () => {
   // NOTE: it's important that the return value is a `Route` component!
-  render() {
-    // In case the application is served by a proxy server, we assume that
-    // the reverse proxy router handles requests forwarding to the specified
-    // service.
-    // For example, if the current "loaded" app is products and I click
-    // on a link to discounts, the products app does not know about the
-    // discount routes, thus falling back to this "catch all route" component.
-    // At this point we force a page reload, effectively handing the
-    // route control logic to the reverse proxy in our cluster. There,
-    // the router mapping will match the discounts route and it will forward
-    // the request to the discounts app.
-    // If no route matches, the application fallback will handle the request
-    // instead, showing e.g. a 404 page.
-    if (this.props.servedByProxy) return <Route component={ForcePageReload} />;
+  const servedByProxy = useApplicationContext(
+    context => context.environment.servedByProxy
+  );
+  // In case the application is served by a proxy server, we assume that
+  // the reverse proxy router handles requests forwarding to the specified
+  // service.
+  // For example, if the current "loaded" app is products and I click
+  // on a link to discounts, the products app does not know about the
+  // discount routes, thus falling back to this "catch all route" component.
+  // At this point we force a page reload, effectively handing the
+  // route control logic to the reverse proxy in our cluster. There,
+  // the router mapping will match the discounts route and it will forward
+  // the request to the discounts app.
+  // If no route matches, the application fallback will handle the request
+  // instead, showing e.g. a 404 page.
+  if (servedByProxy) return <Route component={ForcePageReload} />;
```

Comment:
I somewhat feel strange about this as it would be true if servedByProxy was 'false' (string) which it is when it's set at times I think. Maybe in the future we can expose small hooks encapsulating this. Like `useIsServedByProxy` so we encapsulate that.

Reply:
No we coerce the values in the app provider, so we can safely expect a boolean here

https://github.com/commercetools/merchant-center-application-kit/blob/4819f78a3eb3fb7055d372f8337020980fa8993c/packages/application-shell/src/components/application-shell/application-shell.js#L490-L492

Reply:
Ah forgot about that. Maybe we can simplify some code in the mc-fe then.